harness/determined/cli/workspace.py

```diff
@@ -88,19 +88,27 @@ def list_workspaces(args: Namespace) -> None:
     sess = cli.setup_session(args)
     orderArg = bindings.v1OrderBy[args.order_by.upper()]
     sortArg = bindings.v1GetWorkspacesRequestSortBy[args.sort_by.upper()]
-    internal_offset = args.offset or 0
+    try:
+        original_offset = args.offset or 0
+        internal_offset = args.offset or 0
+        limit = args.limit
+    except AttributeError:
```

Comment:
I really don't like this handling of the error with exceptions, it's quite a code smell. The CLI should work in a way that if we are accessing args, they are definitely present. Could we add some comments pointing out the information that you are providing in the PR? Specifically about this being a hack, and us dealing with args that are being passed from another cli function which doesn't have the same interface? It may help to also point to this PR just so there's no ambiguity.

Reply:
ok I've pushed a commit which un-does these changes and adds intermediate functions such as `print_workspace_projects_list` with specific typed parameters; if either plan looks like the way forward, Web team has asked ML Sys team to resolve